File: 2015/2_2.rb
def volume(l, w, h)
    return l*w*h
end

def perimeter(w,h)
    return 2*w+2*h
end

total_sqft = 0
File.open("2_1_input.txt").each_line do |dimensions|
    l,w,h = dimensions.split("x").map(&:to_i)
    total_sqft +=  [perimeter(l,w), perimeter(w,h), perimeter(l,h)].min + volume(l,w,h)
end
puts total_sqft

File: 2015/2_2.py
def volume(l, w, h):
	return l*w*h

def perimeter(w,h):
	return 2*w+2*h

file = open("2_1_input.txt", "r")

total_sqft = 0
for dimensions in file:
	l,w,h = dimensions.split("x")
	l = int(l)
	w = int(w)
	h = int(h)
	total_sqft = total_sqft + min(perimeter(l,w), perimeter(w,h), perimeter(l,h)) + volume(l,w,h)
print total_sqft